### PR TITLE
Load pattern for dim

### PR DESCRIPTION
**Pull request by hoanguyen071710**

### CODE REVIEW

File: prod_to_s3_load.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+def get_latest_insert_prod(conn):
+    query = """
+    SELECT *
+    FROM User
+    WHERE DATE(updated_at) = CURRENT_DATE() - 1
+    AND status = 'I'
+    """
+    result = conn.execute(text(query)).fetchall()
+    return result
```
Comment (boushphong) [inline prod_to_s3_load.py:31]: We just need one `method` called `get_latest_records`

File: prod_to_s3_load.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def get_latest_snapshot_datalake():
+    snap_dir = get_path_snapshot(1)
```
Comment (boushphong) [inline prod_to_s3_load.py:35]: please change `get_path_snapshot(1)`

**Review by boushphong — APPROVED** [review]

File: ETL/s3_to_dw_load.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    query = f"""
+    INSERT INTO user_dim (
+        user_id, name, email, start_date, end_date, is_current
+    ) 
+    VALUES {update_values}
+    ON CONFLICT (user_id, end_date) DO 
+    UPDATE 
+    SET 
+        end_date = EXCLUDED.start_date - interval '1 second',
+        is_current = false
```
Comment (boushphong) [inline ETL/s3_to_dw_load.py:69]: What is the smallest unit of time for this? is second small enough?
Can it be `interval '1 millisecond'`?

File: ETL/s3_to_dw_load.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+def get_path_snapshot(days):
+    today = str(datetime.datetime.now() - datetime.timedelta(days=days))[0:10].replace(
+        "-", ""
+    )
+    return f"/stock_db/user/partition={today}/user.csv"
+
+
+def element_to_string(element):
+    return f"{element}"
```
Comment (boushphong) [inline ETL/s3_to_dw_load.py:41]: what about str(element) hardcoded?

File: ETL/s3_to_dw_load.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+
+def get_records(df, records_type):
+    if records_type != "All":
+        df = df.loc[df["status"] == records_type]
+    df.rename(columns={"udated_at": "start_date"})
+    df = df.drop(columns=["status"])
+    df["end_date"] = "9999-01-01 00:00:00"
```
Comment (boushphong) [inline ETL/s3_to_dw_load.py:49]: 9999-12-31 23:59:59 to be exact

File: ETL/s3_to_dw_load.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+
+def element_to_string(element):
+    return f"{element}"
+
+
+def get_records(df, records_type):
+    if records_type != "All":
+        df = df.loc[df["status"] == records_type]
+    df.rename(columns={"udated_at": "start_date"})
```
Comment (boushphong) [inline ETL/s3_to_dw_load.py:47]: Grammar `updated_at`

**Comment by boushphong** [comment]
Can you change this to boolean?
https://github.com/Analytics-Ninjas/glorious-data-pipeline/blob/main/db/dwh_schema.sql#L18

**Comment by hoanguyen071710** [comment]
> Can you change this to boolean? https://github.com/Analytics-Ninjas/glorious-data-pipeline/blob/main/db/dwh_schema.sql#L18

Already fix with latest commit in this branch

**Pull request merged**